crates/biome_html_formatter/src/html/auxiliary/self_closing_element.rs

```diff
@@ -12,24 +12,40 @@ impl FormatNodeRule<HtmlSelfClosingElement> for FormatHtmlSelfClosingElement {
             slash_token,
             r_angle_token,
         } = node.as_fields();
+        let bracket_same_line = f.options().bracket_same_line().value();
 
+        write!(f, [l_angle_token.format(), name.format(), space()])?;
+
+        let attr_group_id = f.group_id("element-attr-group-id");
         write!(
             f,
-            [
-                l_angle_token.format(),
-                name.format(),
-                space(),
-                attributes.format(),
-                space(),
-            ]
-        )?;
+            [&group(&format_with(|f| {
+                attributes.format().fmt(f)?;
+
+                // Whitespace sensitivity takes precedence over bracketSameLine for correctness.
+                //
+                // The r_angle is placed inside this group because prettier always includes this token
+                // in the same group as the attributes, unless the token is being borrowed.
+                // When these tokens are borrowed, they are managed by the sibling `HtmlElementList` formatter.
+                if bracket_same_line {
+                    write!(f, [hard_space()])?;
+                } else {
+                    write!(f, [soft_line_break_or_space()])?;
+                }
 
-        if slash_token.is_some() {
-            write!(f, [slash_token.format()])?;
-        } else {
-            write!(f, [text("/")])?;
-        }
+                // TODO: These tokens (the `/>`) are not yet borrowed by sibling elements for whitespace sensitivity.
+                if slash_token.is_some() {
+                    write!(f, [slash_token.format()])?;
+                } else {
+                    write!(f, [text("/")])?;
```

Comment:
NIT: A couple things:
- from the Todo, it's not clear what needs to be done
- if possible, I would avoid adding a slash if users didn't add one. Maybe we can iterate this in another issue/pr